File: solutions/clojure/triangle/1/src/triangle.clj
(ns triangle)

(defn triangle?
  "Returns true if the numbers are sides for valid triangle"
  [a b c]
  (and
    (> (* a b c) 0)
    (> (+ a b) c)
    (> (+ a c) b)
    (> (+ b c) a)
    ))

(defn equilateral?
  "Returns true if the triangle with sides a, b, and c is equilateral;
  otherwise, it returns false."
  [a b c]
  (and (triangle? a b c) (== a b c)))

(defn isosceles?
  "Returns true if the triangle with sides a, b, and c is isosceles;
  otherwise, it returns false."
  [a b c]
  (and (triangle? a b c) (or (== a b) (== b c) (== a c))))

(defn scalene?
  "Returns true if the triangle with sides a, b, and c is scalene;
  otherwise, it returns false."
  [a b c]
  (and (triangle? a b c) (not (isosceles? a b c))))

File: solutions/python/triangle/1/triangle.py
def equilateral(sides):
    return valid_triangle(sides) and (sides[0] == sides[1] == sides[2])


def isosceles(sides):
    a = sides[0]
    b = sides[1]
    c = sides[2] 
    return valid_triangle(sides) and ((a == b) or (b == c) or (a == c))


def scalene(sides):
    a = sides[0]
    b = sides[1]
    c = sides[2]
    return valid_triangle(sides) and (a != b) and (b != c) and (a != c)

def valid_triangle(sides):
    a = sides[0]
    b = sides[1]
    c = sides[2]
    return (a + b > c) and (a + c > b) and (b + c > a)
